core/object.h

```diff
@@ -140,6 +140,21 @@ struct PropertyInfo {
 	PropertyHint hint;
 	String hint_string;
 	uint32_t usage;
+	uint32_t meta;
+
+	enum Metadata {
+		METADATA_NONE,
+		METADATA_INT_IS_I8,
+		METADATA_INT_IS_I16,
+		METADATA_INT_IS_I32,
+		METADATA_INT_IS_I64,
+		METADATA_INT_IS_UI8,
+		METADATA_INT_IS_UI16,
+		METADATA_INT_IS_UI32,
+		METADATA_INT_IS_UI64,
+		METADATA_REAL_IS_FLOAT,
+		METADATA_REAL_IS_DOUBLE
+	};
```

Comment:
@reduz Are you fine with this?

Reply:
Nope, please do the following:

* use a property usage for signed
* use another property usage for 64 bits

There is no point at all in supporting 8 and 16 bits. I prefer this rather than adding yet another field to property info (which is used everywhere), which is just bloat for something so core, and confusing/annoying to anyone reading the code.

Reply:
I can see the point in avoiding the bloat of adding a field to a struct that is used everywhere. What if we surround it with `#ifdef DEBUG_METHODS_ENABLED`?
I can't take the second statement seriously. This code couldn't be simpler. I can't imagine anyone getting confused by it, let alone annoyed.

Reply:
I'm very serious about it, a PR like this won't go through. My first priority will always be to keep bloat out of core and this is an extremely core class. If for you this inconsistency is unacceptable, I prefer to remove the 4 occurrences of the < 32 bits arguments so there are no longer any functions using them.

Reply:
Does that mean surrounding `#ifdef DEBUG_METHODS_ENABLED` is not an option? If not, I'm thinking of an alternative to having the field in `PropertyInfo`.

Reply:
No, please let's not add hack to that. Definitely go the property usage way and let's get rid of 8/16 bit access if you need them.

Reply:
Even modern processors don't really support this type of access, it's a thing of the past.